src/main/java/com/cegeka/academy/service/userChallenge/UserChallengeServiceImpl.java

```diff
@@ -26,4 +30,42 @@
         return userChallengeList.stream().map(userChallenge -> UserChallengeMapper.convertUserChallengeToUserChallengeDTO(userChallenge)).collect(Collectors.toList());
 
     }
+
+    @Override
+    public void updateUserChallengeStatus(Long userChallengeId, String status) throws NotFoundException, InvalidUserChallengeStatusException {
+
+        UserChallenge userChallenge = userChallengeRepository.findById(userChallengeId)
+                .orElseThrow(() -> new NotFoundException().setMessage("User challenge does not exists."));
+
+        if(status.equalsIgnoreCase(UserChallengeStatus.ACCEPTED.toString()) || status.equalsIgnoreCase(UserChallengeStatus.CANCELED.toString())){
+
```

Comment:
In UserChallengeStatus, te rog sa definesti o metoda public static care sa primeasca ca parametru de intrare un String status si care sa returneze un UserChallengeStatus. Scopul metodei este de a extrage valoarea enum-ului pe baza un string. In cazul in care parametrul de intrare este invalid, arunci acolo exceptie. Daca muti logica in UserChallengeStatus ofera avantajul ca codul scris de tine poate sa fie refolosit si de altcineva fara sa mai implementeze.